src/snapshot.js

```diff
@@ -222,11 +203,14 @@ export function restorePlayerSnapshot(player, snapshotObject) {
     // Reace the `canplay` event with a timeout.
     player.one('contentcanplay', tryToResume);
     player.ads.tryToResumeTimeout_ = player.setTimeout(tryToResume, 2000);
-  } else if (!player.ended() || !snapshotObject.ended) {
+  } else {
     // if we didn't change the src, just restore the tracks
     restoreTracks();
-    // the src didn't change and this wasn't a postroll
-    // just resume playback at the current time.
-    player.play();
+
+    if (!player.ended() || !snapshotObject.ended) {
```

Comment:
Do you know why it checks both `player.ended()` and `snapshotObject.ended`? It seems like one should be enough, so a comment would help at least.

Reply:
Looks like it comes from here: https://github.com/videojs/videojs-contrib-ads/commit/a11cbce454c56cbf214a37908ba82bad8fe12a36, it seems like it's to support postrolls?

Reply:
I just left this conditional as it already was, but with that said I can't think of a case where `!player.ended()` and `!snapshotObject.ended` would not have the same value, but I may be missing something.

Reply:
server-side ads could potentially have a snapshot of ended but a player not ended at least at the time of the initial writing. I think we don't do snapshots for that kind of video anymore?

Reply:
I guess we could either leave it as is or launch a detailed investigation and I'm fine with both hah. I just want to make sure we build more clarity around these sorts of details over time.

Reply:
I'm all for a detailed investigation, but I don't think there's any reason it should hold up these bugs getting fixed.

Reply:
I'm going to investigate right now

Reply:
Following investigation, we are going to just check for `!player.ended()` here since if we reach this line the video element was not recycled.